jaggr-core/src/main/java/com/ibm/jaggr/core/impl/layer/LayerImpl.java

```diff
@@ -1002,11 +1061,46 @@ public static ILayerCache newLayerCache(IAggregator aggregator) {
 	}
 
 	protected void setResponseHeaders(HttpServletRequest request, HttpServletResponse response, int size) {
-		response.setContentType("application/x-javascript; charset=utf-8"); //$NON-NLS-1$
+		response.setContentType(
+				RequestUtil.isSourceMapRequest(request) ?
+						"application/json; charset=utf-8" : //$NON-NLS-1$
+						"application/javascript; charset=utf-8"); //$NON-NLS-1$
 		response.setContentLength(size);
 		if (RequestUtil.isGzipEncoding(request)) {
 			response.setHeader("Content-Encoding", "gzip"); //$NON-NLS-1$ //$NON-NLS-2$
 		}
 
 	}
-}
\ No newline at end of file
+
+	/**
+	 * Returns a byte array containing the gzipped contents of the input stream
+	 *
+	 * @param in
+	 *            the input stream to zip
+	 * @return the gzipped contents in a byte array
+	 * @throws IOException
+	 */
+	protected byte[] zip(InputStream in) throws IOException {

```

Comment:
Do these methods belong in https://github.com/OpenNTF/JavascriptAggregator/blob/d8c2af5109784269a9f0d957ac6b243386a8d879/jaggr-core/src/main/java/com/ibm/jaggr/core/util/ZipUtil.java ?